src/http/routes.cc

```diff
@@ -81,8 +81,11 @@ std::unique_ptr<reply> routes::exception_reply(std::exception_ptr eptr) {
 }
 
 future<std::unique_ptr<reply> > routes::handle(const sstring& path, std::unique_ptr<request> req, std::unique_ptr<reply> rep) {
-    handler_base* handler = get_handler(str2type(req->_method),
-            normalize_url(path), req->param);
+    handler_base* handler = _default_handler;
+    handler_base* rule_handler = get_handler(str2type(req->_method), normalize_url(path), req->param);
+    if(rule_handler != nullptr) {
+        handler = rule_handler;
+    }
```

Comment:
The logic of this change is unnecessarily complicated... You can leave the original `handler = get_handler()` code untouched, and just after it add the lines
```c++
if (handler == nullptr) {
    handler = _default_handler
}
```
But even better, the get_handler() function should just handle the default! In the end of the get_handler() function definition just return _default_handler instead of nullptr. That's it!

Reply:
Good suggestion.